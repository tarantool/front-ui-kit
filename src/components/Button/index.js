// @flow
import * as React from 'react';
import { css, cx } from 'emotion'

const styles = {
  btn: css`
    border: solid 1px gray;
  `
}

type ButtonProps = {
  className?: string,
  children?: React.Node,
  onClick?: (MouseEvent) => void
};

export default ({ children, className, ...props }: ButtonProps) => (
  <button className={cx(className, styles.btn)} {...props}>{children}</button>
);
