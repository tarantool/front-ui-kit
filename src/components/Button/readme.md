Button example:

```js
<Button>Click me</Button>
```
