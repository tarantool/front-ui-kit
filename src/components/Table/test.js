import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from '../Button';
import { Table } from './index';

const columns = [
  {
    Header: 'First Name',
    accessor: 'firstName',
    disableSortBy: true
  },
  {
    Header: 'Last Name',
    accessor: 'lastName',
    disableSortBy: true
  },
  {
    Header: 'Age',
    accessor: 'age',
    disableSortBy: true
  },
  {
    Header: 'Visits',
    accessor: 'visits',
    disableSortBy: true
  },
  {
    Header: 'Status',
    accessor: 'status',
    disableSortBy: true
  },
  {
    Header: 'Profile Progress',
    accessor: 'progress'
  }
];

describe('Table', () => {
  it('renders empty', () => {
    const tree = renderer.create(
      <>
        <Table columns={columns} data={[]}/>
        <Table columns={columns}/>
      </>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders simple', () => {
    const data = [
      {
        firstName: 'spiders',
        lastName: 'profession',
        age: 15,
        visits: 55,
        progress: 66,
        status: 'complicated'
      },
      {
        firstName: 'spiders',
        lastName: 'profession',
        age: 15,
        visits: 55,
        progress: 66,
        status: 'complicated'
      },
      {
        firstName: 'profit',
        lastName: 'quiet',
        age: 8,
        visits: 79,
        progress: 9,
        status: 'complicated'
      },
      {
        firstName: 'hat',
        lastName: 'jelly',
        age: 7,
        visits: 56,
        progress: 67,
        status: <Button text='Refresh' size='xs'/>
      }
    ];

    const tree = renderer.create(
      <Table columns={columns} data={data}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with code row', () => {
    const data = [
      {
        firstName: 'spiders',
        lastName: 'profession',
        age: 15,
        visits: 55,
        progress: 66,
        status: 'complicated',
        code: 'some code1'
      },
      {
        firstName: 'spiders',
        lastName: 'profession',
        age: 15,
        visits: 55,
        progress: 66,
        status: 'complicated',
        code: 'some code2'
      },
      {
        firstName: 'profit',
        lastName: 'quiet',
        age: 8,
        visits: 79,
        progress: 9,
        status: 'complicated'
      },
      {
        firstName: 'hat',
        lastName: 'jelly',
        age: 7,
        visits: 56,
        progress: 67,
        status: 'relationship',
        code: 'some code4'
      }
    ];

    const tree = renderer.create(
      <Table columns={columns} data={data} codeRowKey='code'/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with pagination', () => {
    const data = [
      {
        firstName: 'spiders',
        lastName: 'profession',
        age: 15,
        visits: 55,
        progress: 66,
        status: 'complicated',
        code: 'some code1'
      },
      {
        firstName: 'spiders',
        lastName: 'profession',
        age: 15,
        visits: 55,
        progress: 66,
        status: 'complicated',
        code: 'some code2'
      },
      {
        firstName: 'profit',
        lastName: 'quiet',
        age: 8,
        visits: 79,
        progress: 9,
        status: 'complicated'
      },
      {
        firstName: 'hat',
        lastName: 'jelly',
        age: 7,
        visits: 56,
        progress: 67,
        status: 'relationship',
        code: 'some code4'
      }
    ];

    const tree = renderer.create(
      <Table columns={columns} data={data} codeRowKey='code' pagination/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with pagination and selected rows', () => {
    const data = [
      {
        firstName: 'spiders',
        lastName: 'profession',
        age: 15,
        visits: 55,
        progress: 66,
        status: 'complicated',
        code: 'some code1'
      },
      {
        firstName: 'spiders',
        lastName: 'profession',
        age: 15,
        visits: 55,
        progress: 66,
        status: 'complicated',
        code: 'some code2'
      },
      {
        firstName: 'profit',
        lastName: 'quiet',
        age: 8,
        visits: 79,
        progress: 9,
        status: 'complicated'
      },
      {
        firstName: 'hat',
        lastName: 'jelly',
        age: 7,
        visits: 56,
        progress: 67,
        status: 'relationship',
        code: 'some code4'
      }
    ];

    const tree = renderer.create(
      <Table columns={columns} data={data} codeRowKey='code' onSelectedRowsChange={() => {}} pagination/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
