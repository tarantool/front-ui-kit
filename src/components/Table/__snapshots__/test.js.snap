// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders empty 1`] = `
Array [
  <table
    className="css-1sbsopn"
    role="table"
  >
    <thead>
      <tr
        role="row"
      >
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          First Name
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          Last Name
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          Age
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          Visits
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          Status
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          onClick={[Function]}
          role="columnheader"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          title="Toggle SortBy"
        >
          Profile Progress
          <span
            className="css-1x6dtcs"
          >
            <svg
              className="css-1j1sb4k"
              viewBox="0 0 6 4"
            >
              <use
                xlinkHref="#sortUp"
              />
            </svg>
            <svg
              className="css-1j1sb4k"
              viewBox="0 0 6 4"
            >
              <use
                xlinkHref="#sortDown"
              />
            </svg>
          </span>
        </th>
      </tr>
    </thead>
    <tbody
      role="rowgroup"
    >
      <tr>
        <td
          colSpan={6}
        >
          <div
            className="css-1unytjj"
          >
            <svg
              className="css-1mla397"
              viewBox="0 0 64 41"
            >
              <use
                xlinkHref="#empty-box-no-data"
              />
            </svg>
            <span
              className="css-vyyv1w"
            >
              The list is empty
            </span>
          </div>
        </td>
      </tr>
    </tbody>
  </table>,
  <table
    className="css-1sbsopn"
    role="table"
  >
    <thead>
      <tr
        role="row"
      >
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          First Name
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          Last Name
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          Age
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          Visits
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          Status
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          onClick={[Function]}
          role="columnheader"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          title="Toggle SortBy"
        >
          Profile Progress
          <span
            className="css-1x6dtcs"
          >
            <svg
              className="css-1j1sb4k"
              viewBox="0 0 6 4"
            >
              <use
                xlinkHref="#sortUp"
              />
            </svg>
            <svg
              className="css-1j1sb4k"
              viewBox="0 0 6 4"
            >
              <use
                xlinkHref="#sortDown"
              />
            </svg>
          </span>
        </th>
      </tr>
    </thead>
    <tbody
      role="rowgroup"
    >
      <tr>
        <td
          colSpan={6}
        >
          <div
            className="css-1unytjj"
          >
            <svg
              className="css-1mla397"
              viewBox="0 0 64 41"
            >
              <use
                xlinkHref="#empty-box-no-data"
              />
            </svg>
            <span
              className="css-vyyv1w"
            >
              The list is empty
            </span>
          </div>
        </td>
      </tr>
    </tbody>
  </table>,
]
`;

exports[`Table renders simple 1`] = `
<table
  className="css-1sbsopn"
  role="table"
>
  <thead>
    <tr
      role="row"
    >
      <th
        className="css-4oxon4"
        colSpan={1}
        role="columnheader"
        style={
          Object {
            "cursor": undefined,
          }
        }
      >
        First Name
      </th>
      <th
        className="css-4oxon4"
        colSpan={1}
        role="columnheader"
        style={
          Object {
            "cursor": undefined,
          }
        }
      >
        Last Name
      </th>
      <th
        className="css-4oxon4"
        colSpan={1}
        role="columnheader"
        style={
          Object {
            "cursor": undefined,
          }
        }
      >
        Age
      </th>
      <th
        className="css-4oxon4"
        colSpan={1}
        role="columnheader"
        style={
          Object {
            "cursor": undefined,
          }
        }
      >
        Visits
      </th>
      <th
        className="css-4oxon4"
        colSpan={1}
        role="columnheader"
        style={
          Object {
            "cursor": undefined,
          }
        }
      >
        Status
      </th>
      <th
        className="css-4oxon4"
        colSpan={1}
        onClick={[Function]}
        role="columnheader"
        style={
          Object {
            "cursor": "pointer",
          }
        }
        title="Toggle SortBy"
      >
        Profile Progress
        <span
          className="css-1x6dtcs"
        >
          <svg
            className="css-1j1sb4k"
            viewBox="0 0 6 4"
          >
            <use
              xlinkHref="#sortUp"
            />
          </svg>
          <svg
            className="css-1j1sb4k"
            viewBox="0 0 6 4"
          >
            <use
              xlinkHref="#sortDown"
            />
          </svg>
        </span>
      </th>
    </tr>
  </thead>
  <tbody
    role="rowgroup"
  >
    <tr
      className="css-b02ryf"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      role="row"
    >
      <td
        className="css-1cixjav"
        role="cell"
      >
        spiders
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        profession
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        15
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        55
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        complicated
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        66
      </td>
    </tr>
    <tr
      className="css-b02ryf"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      role="row"
    >
      <td
        className="css-1cixjav"
        role="cell"
      >
        spiders
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        profession
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        15
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        55
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        complicated
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        66
      </td>
    </tr>
    <tr
      className="css-b02ryf"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      role="row"
    >
      <td
        className="css-1cixjav"
        role="cell"
      >
        profit
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        quiet
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        8
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        79
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        complicated
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        9
      </td>
    </tr>
    <tr
      className="css-b02ryf"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      role="row"
    >
      <td
        className="css-1cixjav"
        role="cell"
      >
        hat
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        jelly
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        7
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        56
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        <button
          className="css-16hprh3"
          type="button"
        >
          Refresh
        </button>
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        67
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table renders with code row 1`] = `
<table
  className="css-1sbsopn"
  role="table"
>
  <thead>
    <tr
      role="row"
    >
      <th
        className="css-4oxon4"
        colSpan={1}
        role="columnheader"
        style={
          Object {
            "cursor": undefined,
          }
        }
      >
        First Name
      </th>
      <th
        className="css-4oxon4"
        colSpan={1}
        role="columnheader"
        style={
          Object {
            "cursor": undefined,
          }
        }
      >
        Last Name
      </th>
      <th
        className="css-4oxon4"
        colSpan={1}
        role="columnheader"
        style={
          Object {
            "cursor": undefined,
          }
        }
      >
        Age
      </th>
      <th
        className="css-4oxon4"
        colSpan={1}
        role="columnheader"
        style={
          Object {
            "cursor": undefined,
          }
        }
      >
        Visits
      </th>
      <th
        className="css-4oxon4"
        colSpan={1}
        role="columnheader"
        style={
          Object {
            "cursor": undefined,
          }
        }
      >
        Status
      </th>
      <th
        className="css-4oxon4"
        colSpan={1}
        onClick={[Function]}
        role="columnheader"
        style={
          Object {
            "cursor": "pointer",
          }
        }
        title="Toggle SortBy"
      >
        Profile Progress
        <span
          className="css-1x6dtcs"
        >
          <svg
            className="css-1j1sb4k"
            viewBox="0 0 6 4"
          >
            <use
              xlinkHref="#sortUp"
            />
          </svg>
          <svg
            className="css-1j1sb4k"
            viewBox="0 0 6 4"
          >
            <use
              xlinkHref="#sortDown"
            />
          </svg>
        </span>
      </th>
    </tr>
  </thead>
  <tbody
    role="rowgroup"
  >
    <tr
      className="css-14r34si"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      role="row"
    >
      <td
        className="css-1cixjav"
        role="cell"
      >
        spiders
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        profession
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        15
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        55
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        complicated
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        66
      </td>
    </tr>
    <tr
      className="css-y47bfa"
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    >
      <td
        className="css-tqxve2"
        colSpan={6}
      >
        some code1
      </td>
    </tr>
    <tr
      className="css-14r34si"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      role="row"
    >
      <td
        className="css-1cixjav"
        role="cell"
      >
        spiders
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        profession
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        15
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        55
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        complicated
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        66
      </td>
    </tr>
    <tr
      className="css-y47bfa"
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    >
      <td
        className="css-tqxve2"
        colSpan={6}
      >
        some code2
      </td>
    </tr>
    <tr
      className="css-b02ryf"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      role="row"
    >
      <td
        className="css-1cixjav"
        role="cell"
      >
        profit
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        quiet
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        8
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        79
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        complicated
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        9
      </td>
    </tr>
    <tr
      className="css-14r34si"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      role="row"
    >
      <td
        className="css-1cixjav"
        role="cell"
      >
        hat
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        jelly
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        7
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        56
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        relationship
      </td>
      <td
        className="css-1cixjav"
        role="cell"
      >
        67
      </td>
    </tr>
    <tr
      className="css-y47bfa"
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    >
      <td
        className="css-tqxve2"
        colSpan={6}
      >
        some code4
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table renders with pagination 1`] = `
Array [
  <table
    className="css-1sbsopn"
    role="table"
  >
    <thead>
      <tr
        role="row"
      >
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          First Name
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          Last Name
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          Age
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          Visits
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          Status
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          onClick={[Function]}
          role="columnheader"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          title="Toggle SortBy"
        >
          Profile Progress
          <span
            className="css-1x6dtcs"
          >
            <svg
              className="css-1j1sb4k"
              viewBox="0 0 6 4"
            >
              <use
                xlinkHref="#sortUp"
              />
            </svg>
            <svg
              className="css-1j1sb4k"
              viewBox="0 0 6 4"
            >
              <use
                xlinkHref="#sortDown"
              />
            </svg>
          </span>
        </th>
      </tr>
    </thead>
    <tbody
      role="rowgroup"
    >
      <tr
        className="css-14r34si"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        role="row"
      >
        <td
          className="css-1cixjav"
          role="cell"
        >
          spiders
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          profession
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          15
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          55
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          complicated
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          66
        </td>
      </tr>
      <tr
        className="css-y47bfa"
        onClick={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <td
          className="css-tqxve2"
          colSpan={6}
        >
          some code1
        </td>
      </tr>
      <tr
        className="css-14r34si"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        role="row"
      >
        <td
          className="css-1cixjav"
          role="cell"
        >
          spiders
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          profession
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          15
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          55
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          complicated
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          66
        </td>
      </tr>
      <tr
        className="css-y47bfa"
        onClick={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <td
          className="css-tqxve2"
          colSpan={6}
        >
          some code2
        </td>
      </tr>
      <tr
        className="css-b02ryf"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        role="row"
      >
        <td
          className="css-1cixjav"
          role="cell"
        >
          profit
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          quiet
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          8
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          79
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          complicated
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          9
        </td>
      </tr>
      <tr
        className="css-14r34si"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        role="row"
      >
        <td
          className="css-1cixjav"
          role="cell"
        >
          hat
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          jelly
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          7
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          56
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          relationship
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          67
        </td>
      </tr>
      <tr
        className="css-y47bfa"
        onClick={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <td
          className="css-tqxve2"
          colSpan={6}
        >
          some code4
        </td>
      </tr>
    </tbody>
  </table>,
  <div
    className="css-kw88hx"
  >
    <div
      className="css-k008qs"
    >
      <div
        className="css-lf2s8g"
      >
        1
        -
        4
         of 
        4
         items
      </div>
      <div>
        <button
          className="css-1kyfbge"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <svg
            className="css-1hsbuls"
            viewBox="0 0 14 14"
          >
            <use
              xlinkHref="#chevron-up"
            />
          </svg>
        </button>
      </div>
      <div>
        <button
          className="css-1u7s8st"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
      </div>
      <div>
        <button
          className="css-1kyfbge"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <svg
            className="css-oiyldh"
            viewBox="0 0 14 14"
          >
            <use
              xlinkHref="#chevron-up"
            />
          </svg>
        </button>
      </div>
      <div
        className="css-1vkrryt"
        onClick={[Function]}
      >
        <button
          className="css-xhnfgx"
          type="button"
        >
          10 / page 
          <svg
            className="css-ok8b0z"
            viewBox="0 0 14 14"
          >
            <use
              xlinkHref="#chevron-up"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`Table renders with pagination and selected rows 1`] = `
Array [
  <table
    className="css-1sbsopn"
    role="table"
  >
    <thead>
      <tr
        role="row"
      >
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          <div>
            <label
              className="css-759u60"
              title="Toggle All Rows Selected"
            >
              <input
                checked={false}
                className="css-1y12ua6"
                disabled={false}
                onChange={[Function]}
                type="checkbox"
              />
              <div
                className="css-79elbk"
              >
                <svg
                  className="css-1v31moo"
                  viewBox="0 0 16 16"
                >
                  <use
                    xlinkHref="#checkbox"
                  />
                </svg>
              </div>
            </label>
          </div>
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          First Name
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          Last Name
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          Age
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          Visits
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          role="columnheader"
          style={
            Object {
              "cursor": undefined,
            }
          }
        >
          Status
        </th>
        <th
          className="css-4oxon4"
          colSpan={1}
          onClick={[Function]}
          role="columnheader"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          title="Toggle SortBy"
        >
          Profile Progress
          <span
            className="css-1x6dtcs"
          >
            <svg
              className="css-1j1sb4k"
              viewBox="0 0 6 4"
            >
              <use
                xlinkHref="#sortUp"
              />
            </svg>
            <svg
              className="css-1j1sb4k"
              viewBox="0 0 6 4"
            >
              <use
                xlinkHref="#sortDown"
              />
            </svg>
          </span>
        </th>
      </tr>
    </thead>
    <tbody
      role="rowgroup"
    >
      <tr
        className="css-14r34si"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        role="row"
      >
        <td
          className="css-1cixjav"
          role="cell"
        >
          <div>
            <label
              className="css-759u60"
              title="Toggle Row Selected"
            >
              <input
                checked={false}
                className="css-1y12ua6"
                disabled={false}
                onChange={[Function]}
                type="checkbox"
              />
              <div
                className="css-79elbk"
              >
                <svg
                  className="css-1v31moo"
                  viewBox="0 0 16 16"
                >
                  <use
                    xlinkHref="#checkbox"
                  />
                </svg>
              </div>
            </label>
          </div>
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          spiders
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          profession
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          15
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          55
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          complicated
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          66
        </td>
      </tr>
      <tr
        className="css-y47bfa"
        onClick={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <td
          className="css-tqxve2"
          colSpan={7}
        >
          some code1
        </td>
      </tr>
      <tr
        className="css-14r34si"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        role="row"
      >
        <td
          className="css-1cixjav"
          role="cell"
        >
          <div>
            <label
              className="css-759u60"
              title="Toggle Row Selected"
            >
              <input
                checked={false}
                className="css-1y12ua6"
                disabled={false}
                onChange={[Function]}
                type="checkbox"
              />
              <div
                className="css-79elbk"
              >
                <svg
                  className="css-1v31moo"
                  viewBox="0 0 16 16"
                >
                  <use
                    xlinkHref="#checkbox"
                  />
                </svg>
              </div>
            </label>
          </div>
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          spiders
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          profession
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          15
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          55
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          complicated
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          66
        </td>
      </tr>
      <tr
        className="css-y47bfa"
        onClick={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <td
          className="css-tqxve2"
          colSpan={7}
        >
          some code2
        </td>
      </tr>
      <tr
        className="css-b02ryf"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        role="row"
      >
        <td
          className="css-1cixjav"
          role="cell"
        >
          <div>
            <label
              className="css-759u60"
              title="Toggle Row Selected"
            >
              <input
                checked={false}
                className="css-1y12ua6"
                disabled={false}
                onChange={[Function]}
                type="checkbox"
              />
              <div
                className="css-79elbk"
              >
                <svg
                  className="css-1v31moo"
                  viewBox="0 0 16 16"
                >
                  <use
                    xlinkHref="#checkbox"
                  />
                </svg>
              </div>
            </label>
          </div>
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          profit
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          quiet
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          8
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          79
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          complicated
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          9
        </td>
      </tr>
      <tr
        className="css-14r34si"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        role="row"
      >
        <td
          className="css-1cixjav"
          role="cell"
        >
          <div>
            <label
              className="css-759u60"
              title="Toggle Row Selected"
            >
              <input
                checked={false}
                className="css-1y12ua6"
                disabled={false}
                onChange={[Function]}
                type="checkbox"
              />
              <div
                className="css-79elbk"
              >
                <svg
                  className="css-1v31moo"
                  viewBox="0 0 16 16"
                >
                  <use
                    xlinkHref="#checkbox"
                  />
                </svg>
              </div>
            </label>
          </div>
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          hat
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          jelly
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          7
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          56
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          relationship
        </td>
        <td
          className="css-1cixjav"
          role="cell"
        >
          67
        </td>
      </tr>
      <tr
        className="css-y47bfa"
        onClick={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <td
          className="css-tqxve2"
          colSpan={7}
        >
          some code4
        </td>
      </tr>
    </tbody>
  </table>,
  <div
    className="css-kw88hx"
  >
    <div
      className="css-k008qs"
    >
      <div
        className="css-lf2s8g"
      >
        1
        -
        4
         of 
        4
         items
      </div>
      <div>
        <button
          className="css-1kyfbge"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <svg
            className="css-1hsbuls"
            viewBox="0 0 14 14"
          >
            <use
              xlinkHref="#chevron-up"
            />
          </svg>
        </button>
      </div>
      <div>
        <button
          className="css-1u7s8st"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
      </div>
      <div>
        <button
          className="css-1kyfbge"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <svg
            className="css-oiyldh"
            viewBox="0 0 14 14"
          >
            <use
              xlinkHref="#chevron-up"
            />
          </svg>
        </button>
      </div>
      <div
        className="css-1vkrryt"
        onClick={[Function]}
      >
        <button
          className="css-xhnfgx"
          type="button"
        >
          10 / page 
          <svg
            className="css-ok8b0z"
            viewBox="0 0 14 14"
          >
            <use
              xlinkHref="#chevron-up"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
]
`;
